src/plugins/socketmon/socketmon.cpp

```diff
@@ -579,166 +355,14 @@ static event_response_t tcpe_x64_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info
-        goto done;
-
-    // Convert port to little endian
-    tcpl.port = __bswap_16(tcpl.port);
-
-    ctx.addr = tcpl.inetaf;
-    if ( VMI_FAILURE == vmi_read(vmi, &ctx, sizeof(inetaf), &inetaf, NULL) )
-        goto done;
-
-    if ( tcpl.localaddr )
-    {
-        ctx.addr = tcpl.localaddr;
-        if ( VMI_FAILURE == vmi_read(vmi, &ctx, sizeof(local), &local, NULL) )
-            goto done;
-    }
-
-    if ( local.pdata )
-    {
-        ctx.addr = local.pdata;
-        if ( VMI_FAILURE == vmi_read_addr(vmi, &ctx, &p1) )
-            goto done;
-    }
-
-    lip = read_ip_string(vmi, ctx, p1, inetaf.addressfamily);
-    if (!lip) goto done;
-
-    if (!drakvuf_get_process_data(drakvuf, tcpl.owner, &owner_proc_data))
-        goto done;
-
-    print_tcpl_ret(drakvuf, info, s, owner_proc_data, inetaf.addressfamily, lip, tcpl.port);
-
-done:
-    g_free(const_cast<char*>(owner_proc_data.name));
-    g_free(lip);
-    drakvuf_release_vmi(drakvuf);
-    drakvuf_remove_trap(drakvuf, info->trap, free_wrapper);
-
-    return 0;
-}
-
-static event_response_t tcpl_x86_ret_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
-{
-    return tcpl_ret_cb<tcp_listener_x86, inetaf_x86, local_address_x86>(drakvuf, info);
-}
-
-static event_response_t tcpl_x64_ret_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
-{
-    return tcpl_ret_cb<tcp_listener_x64, inetaf_x64, local_address_x64>(drakvuf, info);
-}
-
-static event_response_t tcpl_win10_x64_ret_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
-{
-    return tcpl_ret_cb<tcp_listener_win10_x64, inetaf_win10_x64, local_address_x64>(drakvuf, info);
-}
-
-static event_response_t tcpl_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
-{
-    struct wrapper* w = (struct wrapper*)g_try_malloc0(sizeof(struct wrapper));
-    w->s = (socketmon*)info->trap->data;
-
-    addr_t rsp = 0;
-    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
-    vmi_read_addr_va(vmi, info->regs->rsp, 0, &rsp);
-    drakvuf_release_vmi(drakvuf);
-
-    w->obj = drakvuf_get_function_argument(drakvuf, info, 1);
-
-    if ( !w->obj )
-    {
-        g_free(w);
-        return 0;
-    }
-
-    drakvuf_trap_t* trap = (drakvuf_trap_t*)g_try_malloc0(sizeof(drakvuf_trap_t));
-    trap->breakpoint.lookup_type = LOOKUP_PID;
-    trap->breakpoint.pid = 4;
-    trap->breakpoint.addr_type = ADDR_VA;
-    trap->breakpoint.addr = rsp;
-    trap->type = BREAKPOINT;
-    trap->data = w;
-
-    if (w->s->winver == VMI_OS_WINDOWS_7 || w->s->winver == VMI_OS_WINDOWS_8)
-        trap->cb = ( w->s->pm == VMI_PM_IA32E ) ? tcpl_x64_ret_cb : tcpl_x86_ret_cb;
-    else
-        trap->cb = ( w->s->pm == VMI_PM_IA32E ) ? tcpl_win10_x64_ret_cb : NULL;
-
-    if ( !drakvuf_add_trap(drakvuf, trap) )
-    {
-        printf("Failed to trap return at 0x%lx\n", rsp);
-        g_free(w);
-    }
-
-    return 0;
-}
-
-static event_response_t udpb_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
```

Comment:
Looks to me like you are completely removing the UDP socket monitoring callbacks? Is that intentional?

Reply:
Never mind, from your PR message I see that it is. I would rather leave these parts in, even if they appear broken right now because monitoring UDP sockets did definitely worked for me when I added it. So it might just needs a little tweaking to get it working again. It would be fine to disable the traps for now with a TODO commit added that says the UDP monitoring needs fixing (with also a Github Issue added to track it).

Reply:
OK. I will fix it.